Lib/idlelib/autocomplete.py

```diff
@@ -203,6 +396,14 @@ def fetch_completions(self, what, mode):
                 except OSError:
                     return [], []
 
+            elif mode == DICTKEYS:
+                try:
+                    entity = self.get_entity(what)
+                    keys = entity.keys() if isinstance(entity, dict) else []
+                except:
+                    return [], []
+                smalll = bigl = [k for k in keys if type(k) in (bytes, str)]
```

Comment:
> Bug 3. The keys completion box should only have the dict keys, as with attributes, but not the global list.

@terryjreedy Could we fix this by adding a conditional filter to the list comprehension?

```suggestion
                smalll = bigl = [k for k in keys if type(k) in (bytes, str) and k not in globals()]
```

Reply:
@aeros167, this was done intentionally (see details in separate comment.)

Reply:
I don't see from the snippet given how the 230+ globals are there.  But they should never be in keys.

Reply:
>I don't see from the snippet given how the 230+ globals are there. But they should never be in keys.

@terryjreedy Hmm, this suggestion might be in the wrong location then. This PR was the first time I've read through idlelib/autocomplete.py. Based on my understanding, this was list of potential completions for dict keys, so filtering out any of the keys which were located in ``globals()`` would address bug 3.

@taleinat I made another suggestion in a comment below which performs filtering from ``globals()`` only when the completion is started after an opening quote.